### PR TITLE
New package: crda

### PR DESCRIPTION
**Pull request by thypon**

### CODE REVIEW

File: srcpkgs/crda/files/set-wireless-regdom [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (xtraeme) [inline srcpkgs/crda/files/set-wireless-regdom]: `/bin/bash` -> `/bin/sh`

File: srcpkgs/crda/files/set-wireless-regdom [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+unset WIRELESS_REGDOM
+. /etc/conf.d/wireless-regdom
+[ -n "${WIRELESS_REGDOM}" ] && iw reg set ${WIRELESS_REGDOM}
```
Comment (xtraeme) [inline srcpkgs/crda/files/set-wireless-regdom]: You are using the `iw` command but the pkg is not depending on it.

File: srcpkgs/crda/template [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+build_style=gnu-makefile
+hostmakedepends="libressl-openssl python python-M2Crypto pkg-config"
+makedepends="libnl-devel libgcrypt-devel"
+short_desc="Central Regulatory Domain Agent for wireless networks"
+maintainer="Andrea Brancaleoni <miwaxe@gmail.com>"
+license="custom"
+homepage="http://wireless.kernel.org/en/developers/Regulatory/CRDA"
+distfiles="https://kernel.org/pub/software/network/crda/${pkgname}-${version}.tar.xz https://www.kernel.org/pub/software/network/wireless-regdb/${_dbname}-${_dbversion}.tar.xz"
+checksum="43fcb9679f8b75ed87ad10944a506292def13e4afb194afa7aa921b01e8ecdbf 438d7f3d62686bc997098d17fe1aff95c6f6ec061aaab90ab7c2c17e8451ce85"
+CFLAGS="-fPIC"
```
Comment (xtraeme) [inline srcpkgs/crda/template]: Please remove those gratuitous vars and use `do_{build,install}` instead.

Comment (thypon) [inline srcpkgs/crda/template]: You intend _dbname/_dbversion? The two packages are tightly coupled there is no sense in packaging them separatly + there is a circular dependencies between them. wireless-regdb makedepends on crda while crda depends on wireless-regdb.

Comment (xtraeme) [inline srcpkgs/crda/template]: Was just talking about CFLAGS and LDFLAGS, the others makes sense for exactly this case.

File: srcpkgs/crda/template [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+homepage="http://wireless.kernel.org/en/developers/Regulatory/CRDA"
+distfiles="https://kernel.org/pub/software/network/crda/${pkgname}-${version}.tar.xz https://www.kernel.org/pub/software/network/wireless-regdb/${_dbname}-${_dbversion}.tar.xz"
+checksum="43fcb9679f8b75ed87ad10944a506292def13e4afb194afa7aa921b01e8ecdbf 438d7f3d62686bc997098d17fe1aff95c6f6ec061aaab90ab7c2c17e8451ce85"
+
+build_options="wardrive"
+desc_option_wardrive="Disable CRDA checks for wardriving competitions"
+
+
+if [ "$build_option_wardrive" ]; then
+	post_extract() {
```
Comment (xtraeme) [inline srcpkgs/crda/template]: Do this inversely:

```
post_extract() {
 if [ "$build_opt" ]; then
    ...
  fi
}
```

Comment (xtraeme) [inline srcpkgs/crda/template]: `/etc/conf.d/wireless-regdb` is a configuration file and is not declared in `conf_files`.

Also I'd like to change its path to just `/etc/wireless-regdb`

File: srcpkgs/crda/template [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+do_build() {
+	cd "${_dbsrc}"
+	make
+	cp *.pem "${wrksrc}/pubkeys"
+
+	cd "${wrksrc}"
+
+	make CC="$CC" CXX="$CXX" LD="$LD" AR="$AR" RANLIB="$RANLIB" \
```
Comment (xtraeme) [inline srcpkgs/crda/template]: CC=$CC is really enough.... no need to duplicate gnu-makefile.

File: srcpkgs/crda/template [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+		 sed -i 's/([0-9][0-9])/(33)/g' db.txt # Max 2W =~ 33dBm
+	}
+fi
+
+do_build() {
+	cd "${_dbsrc}"
+	make
+	cp *.pem "${wrksrc}/pubkeys"
+
+	cd "${wrksrc}"
```
Comment (xtraeme) [inline srcpkgs/crda/template]: And no need for quoting... this is not a PKGBUILD.

File: srcpkgs/crda/template [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+	make CC="$CC" CXX="$CXX" LD="$LD" AR="$AR" RANLIB="$RANLIB" \
+		CPP="$CPP" AS="$AS" OBJDUMP="$OBJDUMP" STRIP="true" all_noverify
+}
+
+do_install() {
+	make DESTDIR="${DESTDIR}" SBINDIR="/usr/sbin" UDEV_RULE_DIR="/usr/lib/udev/rules.d/" install
```
Comment (xtraeme) [inline srcpkgs/crda/template]: `/usr/sbin` -> `/usr/bin`

File: srcpkgs/crda/template [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+_dbname=wireless-regdb
+_dbversion=2015.01.30
+_dbsrc="${XBPS_BUILDDIR}/${_dbname}-${_dbversion}/"
+revision=2
+hostmakedepends="libressl-openssl python python-M2Crypto pkg-config"
+makedepends="libnl-devel libgcrypt-devel"
+depends="iw"
+short_desc="Central Regulatory Domain Agent for wireless networks"
+maintainer="Andrea Brancaleoni <miwaxe@gmail.com>"
+license="custom"
```
Comment (xtraeme) [inline srcpkgs/crda/template]: I'd like to see something else than `custom`.

File: srcpkgs/python-M2Crypto/template [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'python-M2Crypto'
+_mname=M2Crypto
+pkgname="python-${_mname}"
+version=0.22.3
+revision=2
```
Comment (xtraeme) [inline srcpkgs/python-M2Crypto/template]: `revision=1` for new pkgs

File: srcpkgs/crda/template [hunk 10]
```diff
@@ -0,0 +1,57 @@
+# Template file for 'crda'
+pkgname=crda
+version=3.18
+_dbname=wireless-regdb
+_dbversion=2015.01.30
+_dbsrc="${XBPS_BUILDDIR}/${_dbname}-${_dbversion}/"
+revision=2
```
Comment (xtraeme) [inline srcpkgs/crda/template]: revision=1 for new pkgs.

File: srcpkgs/crda/template (hunk 7 above)
Comment (thypon) [inline srcpkgs/crda/template]: I'm not sure what goes on sbin and what in bin on voidlinux. Could we someday use the unified /usr/bin /usr/sbin layout like in fedora/archlinux?

File: srcpkgs/python-M2Crypto/template (hunk 9 above)
Comment (thypon) [inline srcpkgs/python-M2Crypto/template]: rebase when the package is ok.

File: srcpkgs/crda/template (hunk 6 above)
Comment (thypon) [inline srcpkgs/crda/template]: What happens if there is a space in a wrksrc for example?

**Comment by thypon** [comment]
@xtraeme i'm waiting your :+1: then i rebase.

File: srcpkgs/crda/files/set-wireless-regdom [hunk 11]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+unset WIRELESS_REGDOM
+. /etc/conf.d/wireless-regdom
```
Comment (xtraeme) [inline srcpkgs/crda/files/set-wireless-regdom]: `/etc/wireless-regdom`

**Comment by xtraeme** [comment]
@thypon this one looks good enough, thanks :+1:

**Comment by thypon** [comment]
alredy rebased :-)

**Pull request merged**